keps/sig-apps/3715-elastic-indexed-job/kep.yaml

```diff
@@ -0,0 +1,38 @@
+title: Elastic Indexed Job
+kep-number: 3715
+authors:
+  - "@ahg-g"
+owning-sig: sig-apps
+status: implementable
+creation-date: 2023-01-10
+reviewers:
+  - "@alculquicondor"
+approvers:
+  - "@soltysh"
+
+see-also:
+  - "/keps/sig-apps/2214-indexed-job"
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.27"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "1.27"
+  beta: "1.28"
+  stable: "v1.29"
+
+feature-gates:
+  - name: ElasticIndexedJob
+    components:
+      - kube-apiserver
```

Comment:
`kube-controller-manager` should also be thought about the new feature gate, no?

Reply:
> I don't think you we need it in kube-controller-manager because the only change is a refactor to treat both enums as one, and we need that to always be on to handle the new enum value on downgrades.

IMO, you are introducing changes to the controller and they should be similarly gated as anything else.

Reply:
> IMO, you are introducing changes to the controller and they should be similarly gated as anything else.

The changes are not strictly introducing new behavior and guarding it actually would not be safer.

I think of it this way: if one sends a PR now to introduce a function named `IsIndexed(job v1.Job)` and use that instead of explicitly checking `spec.completions=Indexed`, would we consider this a change that needs guarding? I think not. The followup change that pertains to handling this new enum would be to have `IsIndexed` also look at `ElasticIndexed` value, which I think should always be done to safely handle downgrades.

Reply:
Let's see which path we'll use before deciding if this is a new feature, like you said in the other comment.

Reply:
Based on the latest proposal, there will be no changes to the controller, so resolving this comment.